internal/exec/vendor_utils.go

```diff
@@ -494,23 +523,32 @@ func shouldSkipSource(s *schema.AtmosVendorSource, component string, tags []stri
 	return (component != "" && s.Component != component) || (len(tags) > 0 && len(lo.Intersect(tags, s.Tags)) == 0)
 }
 
-func determineSourceType(uri *string, vendorConfigFilePath string) (bool, bool, bool) {
-	// Determine if the URI is an OCI scheme, a local file, or remote
+func determineSourceType(uri *string, vendorConfigFilePath string) (bool, bool, bool, bool) {
+	// Determine if the URI is an OCI scheme, a local file, a remote GitHub source, or a generic remote
 	useOciScheme := strings.HasPrefix(*uri, "oci://")
 	if useOciScheme {
 		*uri = strings.TrimPrefix(*uri, "oci://")
 	}
 
 	useLocalFileSystem := false
 	sourceIsLocalFile := false
+	isGitHubSource := false
+
 	if !useOciScheme {
-		if absPath, err := u.JoinAbsolutePathWithPath(vendorConfigFilePath, *uri); err == nil {
-			uri = &absPath
-			useLocalFileSystem = true
-			sourceIsLocalFile = u.FileExists(*uri)
+		if strings.Contains(*uri, "github.com") {
```

Comment:
I think we should be a little bit more secure, and only check the hostname of a parsed URL. Otherwise we leak the `GITHUB_TOKEN`

Reply:
We should do this: https://github.com/cloudposse/atmos/pull/871/files#diff-3510d27050a93bc3e6375f141c5fded181a42d20457329fbe89b37bdba654563R253

Reply:
done